src/client/client.js

```diff
@@ -1,11 +1,13 @@
 import {render} from 'react-dom';
 import React from 'react';
-import {syncHistory, routeReducer} from 'redux-simple-router'
+import {syncHistory, routeReducer} from 'redux-simple-router';
 import {Map, fromJS} from 'immutable';
 import {ensureState} from 'redux-optimistic-ui';
 
-const makeStore = __PRODUCTION__ ? require('./makeStore.prod.js') : require('./makeStore.dev.js');
-const Root = __PRODUCTION__ ? require('./Root.prod.js') : require('./Root.dev.js');
+const ENV_SUFFIX = __PRODUCTION__ ? 'prod' : 'dev';
+
+const makeStore = require(`./makeStore.${ENV_SUFFIX}.js`);
```

Comment:
Same as before, as long as it's a hardcoded string inside the require
you're good to go. Since __PRODUCTION__ is a webpack var it can eval that
during the build without the runtime

On Wed, Feb 17, 2016, 4:38 PM Alan R. Soares <notifications@github.com>
wrote:

> In src/client/client.js
> <https://github.com/mattkrick/meatier/pull/74#discussion_r53235544>:
>
> >  import {Map, fromJS} from 'immutable';
> >  import {ensureState} from 'redux-optimistic-ui';
> >
> > -const makeStore = __PRODUCTION__ ? require('./makeStore.prod.js') : require('./makeStore.dev.js');
> > -const Root = __PRODUCTION__ ? require('./Root.prod.js') : require('./Root.dev.js');
> > +const ENV_SUFFIX = __PRODUCTION__ ? 'prod' : 'dev';
> > +
> > +const makeStore = require(`./makeStore.${ENV_SUFFIX}.js`);
>
> @mattkrick <https://github.com/mattkrick> What do you suggest?
> The dynamic require was already there, plus those modules are fairly small
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mattkrick/meatier/pull/74/files#r53235544>.
>